docs/IntegrationTest.md

```diff
@@ -0,0 +1,63 @@
+<!--
+ Copyright 2019 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ compliance with the License. You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under the License
+ is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ implied. See the License for the specific language governing permissions and limitations under the
+ License.
+-->
+# Integration Tests
+
+## Jenkins on GKE with helm
+Follow Google Cloud Solutions'
+[tutorial](https://cloud.google.com/solutions/jenkins-on-kubernetes-engine-tutorial) for setting up 
+Jenkins on GKE.
+
+<!--
+TODO(stephenashank): Add another step for the CasC config when we are able to pre-load credentials
+through the helm chart after refactoring the Google Oauth Plugin.
+-->
+Before running helm install, add the following modification to the file `jenkins/values.yaml`:
+
+* Directly below the line which reads `Agent:`, add these three lines:
```

Comment:
@viglesiasce Let me know your thoughts on this. I haven't had great success with using both a file and inline config arguments, so I opted for modifying the cloned values file instead. I think this would be preferred over using this plugin-specific agent image in the general Jenkins on GKE tutorial.

By running through that tutorial with this modification I was able to use the web UI, upload credentials, set up a build, and then then have the agent pod pull this image and executed the kubectl commands to deploy through this plugin.

How long do you think it will take to have `jenkins/values.yaml` with the new oauth plugin version? I might need to add a note about that here depending on if that happens before or after this change is finished.

Reply:
I can get that change in today.

I think its fine to have them change that value. The other option is to have them run an upgrade as follows:

```shell
helm upgrade cd stable/jenkins --reuse-values -f your-changes.yaml
```

Reply:
That is pretty helpful, I think that would make this set up less prone to potential errors from manually editing the file so I'll probably go with this.

Reply:
After looking into this while testing the Privileged flag, I ran into what appears to be a system of this issue: https://github.com/helm/helm/issues/2735. Effectively, the image that appears in the kubernetes cloud configuration isn't changed to the new image and tag. For now I suppose manipulating the file is fine.

Reply:
So in the end since we ship an image with the `lts` tag, all we need is for users to do:
```shell
helm upgrade cd stable/jenkins --reuse-values --set Agent.Image=gcr.io/jenkins-gke-plugin/jenkinsagent
```

Also its probably best to use a global registry endpoint `gcr.io` rather than a regional one `us.gcr.io`.

Reply:
I take that back I was thinking that the default image tag is `lts` but that is only for the Master.Image. We could push versioned ones that match the versions in the chart but that might be too much complexity for the integration test workload.